### ISSUE TITLE
How to predict all photos in a folder

### ISSUE THREAD

**Issue by qa1511** [Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0#18]
Hi, 
Thanks for the code.But I have a question.I noticed in mask-rcnn-prediction.py that he seems to be able to make predictions for only one photo?How about getting him to batch predict? I mean I have more than one photo to predict, I have hundreds to predict, can I get him to predict all the photos in a folder?

**Comment by jwd222** [comment]
> Hi, Thanks for the code.But I have a question.I noticed in mask-rcnn-prediction.py that he seems to be able to make predictions for only one photo?How about getting him to batch predict? I mean I have more than one photo to predict, I have hundreds to predict, can I get him to predict all the photos in a folder?

You can loop through the images in your folder and run the lines 34 to 50 in the `mask-rcnn-prediction.py` file for each image. That'll give you predictions for all those images separately.

**Comment by Kamlesh364** [comment]
@qa151 and @jwd222 thanks for the comments and queries. 

there is a simpler way to run inference on multiple images. Please explore [model.MRCNN.detect](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/model.py#L2482) function. I hope this will help you optimize your process.

**Comment by jwd222** [comment]
> @qa151 and @jwd222 thanks for the comments and queries.
> 
> there is a simpler way to run inference on multiple images. Please explore [model.MRCNN.detect](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/model.py#L2482) function. I hope this will help you optimize your process.

hi thank you for pointing out this method, but are there some constraints to how many images can be passed at a time? As I saw that the `BATCH_SIZE` that is calculated based on `IMAGES_PER_GPU` in (config.py)[https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/config.py#L216C14-L216C25] which is in turn calculated depending on GPU memory and image dimension (as seen here)[https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/config.py#L34].
Your feedback will be highly appreciated.

**Comment by jwd222** [comment]
> @qa151 and @jwd222 thanks for the comments and queries.
> 
> there is a simpler way to run inference on multiple images. Please explore [model.MRCNN.detect](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/model.py#L2482) function. I hope this will help you optimize your process.

hi thank you for pointing out this method, but are there some constraints to how many images can be passed at a time? As I saw that the `BATCH_SIZE` that is calculated based on `IMAGES_PER_GPU` and `GPU_COUNT` in [config.py](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/config.py#L216C14-L216C25) which is in turn calculated depending on GPU memory and image dimension [as seen here](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0/blob/19f195123d6ef2bdb1c5bcf0931757728c09794d/mrcnn/config.py#L34).
Your feedback will be highly appreciated.

**Comment by Kamlesh364** [comment]
Yes, batch_size depends on the size of your input image and GPU memory. But it should not matter how many images you pass during inference as the model takes batch_size equal to the size defined during training, by-default.
